tests/chainermn_tests/links_tests/test_multi_node_chain_list.py

```diff
@@ -279,11 +282,13 @@ def test_cycle_model_gpu(param):
 def check_crossing_model(gpu, param):
     communicator, rank_next, rank_prev = create_communicator(gpu)
 
-    n, d = 100, 10
-    X = np.random.randn(n, d).astype(param.dtype)
-    Y = (np.random.rand(n) * 2).astype(np.int32)
-
     with chainer.using_config('dtype', param.dtype):
+        n, d = 100, 10
+        io_dtype = chainer.get_dtype(map_mixed16=np.float16)
+
+        X = np.random.randn(n, d).astype(io_dtype)
+        Y = (np.random.rand(n) * 2).astype(np.int32)
```

Comment:
Why are these lines moved here?

Reply:
The chainer.get_type needs to be used under a context